From: Haavard Skinnemoen <haavard.skinnemoen@atmel.com>
Date: Tue, 27 May 2008 07:37:42 +0000 (+0200)
Subject: avr32: Fix cpufreq oops when ondemand governor is default
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fhskinnemoen%2Favr32-2.6.git;a=commitdiff_plain;h=f04d264afc51acdffeba9cdf3baf04116687680c

avr32: Fix cpufreq oops when ondemand governor is default

Move the AP7 cpufreq init to late_initcall() so that we don't try to
bring up cpufreq until the governor is ready. x86 also uses
late_initcall() for this.

Signed-off-by: Haavard Skinnemoen <haavard.skinnemoen@atmel.com>
---

diff --git a/arch/avr32/mach-at32ap/cpufreq.c b/arch/avr32/mach-at32ap/cpufreq.c
index 235524b..5dd8d25 100644
--- a/arch/avr32/mach-at32ap/cpufreq.c
+++ b/arch/avr32/mach-at32ap/cpufreq.c
@@ -108,5 +108,4 @@ static int __init at32_cpufreq_init(void)
 {
 	return cpufreq_register_driver(&at32_driver);
 }
-
-arch_initcall(at32_cpufreq_init);
+late_initcall(at32_cpufreq_init);
